Add support for alpha versions

s1monw opened the pull request:
Elasticsearch 5.0 will come with alpha versions which is not supported
in the current version scheme. This commit adds support for aplpha starting
with es 5.0.0 in a backwards compatible way.

s1monw committed:
Add support for alpha versions

Elasticsearch 5.0 will come with alpha versions which is not supported
in the current version scheme. This commit adds support for aplpha starting
with es 5.0.0 in a backwards compatible way.

nik9000 commented on core/src/main/java/org/elasticsearch/Version.java:
```diff
@@ -35,8 +35,8 @@
-    // The logic for ID is: XXYYZZAA, where XX is major version, YY is minor version, ZZ is revision, and AA is Beta/RC indicator
-    // AA values below 50 are beta builds, and below 99 are RC builds, with 99 indicating a release
+    // The logic for ID is: XXYYZZAA, where XX is major version, YY is minor version, ZZ is revision, and AA is alpha/beta/rc indicator
+    // AA values below 25 are fro alpha builder (since 5.0), and above 25 and below 50 are beta builds, and below 99 are RC builds, with 99 indicating a release
```
s/fro/for/

nik9000 commented on core/src/main/java/org/elasticsearch/Version.java:
```diff
@@ -153,11 +154,17 @@ public static Version fromString(String version) {
+                if (buildStr.startsWith("alpha")) {
+                    assert rawMajor >= 5 : "major must be >= 5 but was " + major;
+                    build = Integer.parseInt(buildStr.substring(5));
+                    assert build < 25 : "expected a beta build but " + build + " >= 25";
```
the error message for this seems wrong.

nik9000 replied:
or not. Closing.

nik9000 commented:
LGTM

s1monw committed:
Remove SNAPSHOT from versions in plugin descriptors

We removed leniencey from version parsing which caught problems with
-SNAPSHOT suffixes on plugin properies. This commit removes the -SNAPSHOT
from both es and the extension version and adds tests to ensure we can
parse older versions that allowed -SNAPSHOT in BWC way.

s1monw committed:
fix typo

s1monw commented:
@nik9000 I had some failures due to some old build plugin infos.. I fixed the plugin version, @rjernst can you check my gradle changes?

rjernst commented:
Looks good
